src/Controls/src/Xaml/ApplyPropertiesVisitor.cs

```diff
@@ -579,17 +579,7 @@ static bool TrySetValue(object element, BindableProperty property, bool attached
 					}
 				};
 			else
-				minforetriever = () =>
-				{
-					try
-					{
-						return property.DeclaringType.GetRuntimeProperty(property.PropertyName);
-					}
-					catch (AmbiguousMatchException e)
-					{
-						throw new XamlParseException($"Multiple properties with name '{property.DeclaringType}.{property.PropertyName}' found.", lineInfo, innerException: e);
-					}
-				};
+				minforetriever = () => property.DeclaringType.GetRuntimeProperties().FirstOrDefault(pi => pi.Name == property.PropertyName);
```

Comment:
In case anyone else is also curious, initially I was a bit unsure if using the GRP method would fix the issue of retrieving the correct property when there are multiple present. But after talking to @StephaneDelcroix offline and looking at some docs, GRP is built on top of [GetProperties](https://github.com/microsoft/referencesource/blob/51cf7850defa8a17d815b4700b67116e3fa283c2/mscorlib/system/reflection/RuntimeReflectionExtensions.cs#L25C25-L25C38), and starting from .NET 7, the order in which the properties are returned is "deterministic based upon the metadata ordering in the assembly" ([docs](https://learn.microsoft.com/en-us/dotnet/api/system.type.getproperties?view=net-7.0&redirectedfrom=MSDN#System_Type_GetProperties_System_Reflection_BindingFlags_)). So, using FirstOrDefault in conjunction with GRP works in our favor to get the relevant property we are seeking.